### PR TITLE
SkipIf Enhancements

### PR DESCRIPTION
**Pull request by yuce**
* Moved `it.SkipIf` to `skip.If` and deprecated `it.SkipIf`.
* Added `race`, `slow,` ssl` and `flaky` conditions to `skip.If`.
* Added the `~` operator for versions, which does a loose version comparison.
* Added `it.MarkSlow` and `it.MarkFlaky` to replace `t.SkipNow` and `t.Skipf` calls.
* Added `it.MarkRacy` to mark known tests with race conditions.

### CODE REVIEW

**Comment by utku-caglayan** [comment]
Note: With these changes, all the tests that are marked as "flaky", "slow", and "race" are excluded from the pipeline.
Please correct me if this is wrong

File: internal/skip/skip_test.go [hunk 1]
```diff
@@ -101,6 +122,14 @@ func TestSkipIf(t *testing.T) {
 		skips("hz != 1"),
 		skips("hz != 1.3"),
 		skips("hz != 2"),
+		skips("hz ~ 5"),
```
Comment (utku-caglayan) [inline internal/skip/skip_test.go:125]: Can you explain why this should skip?

File: sql/statement_test.go [hunk 2]
```diff
@@ -67,6 +59,16 @@ func TestStatement_SetCursorBufferSize(t *testing.T) {
 	}
 }
 
+func TestStatement_SetCursorBufferSize_ExpectSigned32Bit(t *testing.T) {
+	skip.If(t, "arch ~ 32bit")
+	v := math.MaxInt32
+	value := v + 1
+	errString := "setting cursor buffer size: signed 32-bit integer number expected: 2147483648: illegal argument error"
```
Comment (utku-caglayan) [inline sql/statement_test.go:66]:
```suggestion
	errString := "setting cursor buffer size: integer overflows int32: illegal argument error"error"
```

Comment (yuce) [reply sql/statement_test.go]: Fixed at: https://github.com/hazelcast/hazelcast-go-client/pull/837/commits/e89081cd76fe42e16d20060e6805b5d21c430603

**Review by utku-caglayan — APPROVED** [review]
Looks great, thanks!

File: internal/skip/skip.go [hunk 3]
```diff
@@ -0,0 +1,536 @@
...
+! operator negates the value of the key.
+
+Many Conditions
+
+More than one condition may be specified by separating them with commas.
+All conditions should be satisfied to skip.
+
+	skip.If(t, "ver > 1.1, hz = 5, os != windows, !enterprise")
+
+You can use multiple SkipIf statements to skip if one one of the conditions is satisfied:
```
Comment (nevzatseferoglu) [inline internal/skip/skip.go:173]:
```suggestion
You can use multiple 'skip.If' statements to skip when one of the conditions is satisfied:
```

**Review by nevzatseferoglu — COMMENTED** [review]
Looks great, I left a comment.

File: internal/skip/skip.go (hunk 3 above)
Comment (yuce) [reply internal/skip/skip.go]: Updated at: https://github.com/hazelcast/hazelcast-go-client/pull/837/commits/0381746202bfbf49acff7086f48d9d75a427de91

**Review by utku-caglayan — DISMISSED** [review]

**Review by nevzatseferoglu — DISMISSED** [review]

**Review by utku-caglayan — APPROVED** [review]

**Review by nevzatseferoglu — DISMISSED** [review]

**Review by nevzatseferoglu — APPROVED** [review]

**Review by utku-caglayan — APPROVED** [review]

**Pull request merged**